### ISSUE TITLE
How coherent should we expect the shakespeare to be?

### ISSUE THREAD

**Issue by Sazazezer** [karpathy/nanoGPT#234]
The examples Andrej give in his output seem to be pretty coherent. I'm not expecting a full brand new work of Shakespeare, but i guess i was expecting consistency between lines.

I got the following final results running shakespeare_char on a google collab that i stitched together (GPU runtime: Intel Xeon CPU @2.20 GHz, 13 GB RAM, Tesla K80 accelerator, and 12 GB GDDR5 VRAM)

step 1000: train loss 0.6846, val loss 1.2164
iter 1000: loss 0.8333, time 16762.25ms, mfu 3.26%

But the output is this:

ANTONY:
Why my lord,
My love, thou havest subtle stifly in the hand,
That such beseech thee that we form these fortunes so dear,
And these years be boon for sight of her chill,
In great time the many prevail
What me, there prose in much servitude for you.

Is this the level i should be expecting, or would further tweaking produce something a little more coherent?

**Comment by dkobak** [comment]
I don't know how you could get to validation loss 1.22 on shakespeare_char -- Andrij got 1.47 (see README) and many people reported the same in various issues.

But with the ~1.47 validation loss, I was getting similar kind of output -- basically complete gibberish when one reads more than several words.

**Comment by dkobak** [comment]
It doesn't matter for the output, but if you really managed to push the validation loss down to 1.22 then I would be SUPER curious to know which parameters you tweaked to achieve that!!

**Comment by dkobak** [comment]
I could not get validation loss below 1.46 by changing any of the basic model size parameters:
```
n_layer = 6
n_head = 6
block_size = 256
n_embd = 384
```
Your 1.22 sounds miles away from what I get.

**Comment by dkobak** [comment]
Do you remember which parameters you were tweaking? 🙏

**Comment by Sazazezer** [comment]
Here's my colab. I haven't really had time to try replicating what i did to get that value, but i've left values in a varying state, so if that helps you at all you're welcome to it. If i ever replicate that value i'll let you know.

https://colab.research.google.com/drive/1j5wfLf3XEmH75INPhx3ksCcuT5soM67A?usp=sharing

**Comment by dkobak** [comment]
Thanks. I now see that you used full Shakespeare text from Project Gutenberg for this experiment, so this may explain the difference in the validation loss. Your version has ~3x larger training data compared to Karpathy's tiny-shakespeare dataset, and the validation set is different.
